kernel/src/hil/hmac.rs

```diff
@@ -0,0 +1,39 @@
+//! Interface for HMAC
+
+use crate::returncode::ReturnCode;
+
+/// Implement this trait and use `set_client()` in order to receive callbacks.
+pub trait Client<'a> {
+    // On success the `result` will contain the digest. On error it will contain
+    // the HMAC error value.
+    fn hash_done(&'a self, result: Result<&'a mut [u8], u32>);
+}
+
+pub const HMAC_KEY_SIZE: usize = 16;
+
+pub trait HMAC<'a> {
+    /// Enable the HMAC hardware.
+    /// Must be called before any other methods
+    fn init(&self);
+
+    /// Disable the HMAC hardware
+    fn deinit(&self);
+
+    /// Set the client instance which will receive `hash_done()` callbacks
+    fn set_client(&'a self, client: &'a dyn Client<'a>);
+
+    /// Set the encryption key.
+    /// Returns `EINVAL` if length is not `HMAC_KEY_SIZE`
```

Comment:
AFAICT this is not what the implementation above is actually doing. The size is never checked.

Reply:
Fixed